staging/src/k8s.io/apimachinery/pkg/util/yaml/decoder.go

```diff
@@ -98,7 +99,7 @@ func (d *YAMLToJSONDecoder) Decode(into interface{}) error {
 	}
 
 	if len(bytes) != 0 {
-		err := yaml.Unmarshal(bytes, into)
```

Comment:
this is `sigs.k8s.io/yaml`. Kubernetes depends on the features that `sigs.k8s.io/yaml` implements on top of `go-yaml`. We cannot simply change `sigs.k8s.io/yaml` to `go-yaml` v3.

Please consider moving forward the `go-yaml` v3 conversation in `sigs.k8s.io/yaml`. Some background:
- https://github.com/kubernetes-sigs/yaml/issues/18
- https://github.com/kubernetes-sigs/yaml/pull/26

/close